### PR TITLE
Pattern page copy edits

### PR DESCRIPTION
**Pull request by mikearildbrown**
This PR:

- Cleans up spelling/grammar issues
- Fixes capitalisation for page names
- Builds in AI assistant suggestions
- Lightly edits headers for consistency/in line with our likely final doc structure
- Ensures all patterns follow the Salt style guide

This PR does NOT:

- Change broader page structures to meet the new standards. This will come later.
- Make any image changes

### CODE REVIEW

**Review by origami-z — APPROVED** [review]

**Comment by origami-z** [comment]
No preview available, not sure what's the problem of the build

**Review by joshwooding — APPROVED** [review]

File: site/docs/patterns/button-bar.mdx [hunk 1]
```diff
@@ -148,11 +148,10 @@ There are two ways to position the button bar:
 
 ### Fixed
 
-The button bar is fixed at the bottom of the viewport when:
+Fix the button bar at the bottom of the viewport when:
 - A user can proceed (or step back) regardless of task completion.
-- If content overflows, button bar has a secondary border style.
 - The task is in a wizard.
-
+If content overflows, the button bar uses a secondary border style.
```
Comment (origami-z) [inline site/docs/patterns/button-bar.mdx:154]: Do you want it to be on the 2nd bullet (in preview) or a separate paragraph?

File: site/docs/patterns/contact-details.mdx [hunk 2]
```diff
@@ -108,14 +108,14 @@ If you’re displaying supplementary details, you can configure them to be colla
   />
 
 <ExampleContainer type="positive" customPillText="Best practices">
-- Consider making supplementary details collapsible when they contain secondary information, and don’t need to be on display at all times.
+Consider making supplementary details collapsible when they contain secondary information, and don’t need to be always on display.
```
Comment (origami-z) [inline site/docs/patterns/contact-details.mdx:111]: Best practice box above this one still has a bullet?

File: site/docs/patterns/forms.mdx [hunk 3]
```diff
@@ -408,10 +411,10 @@ There are two ways to position the button bar:
     contentPosition="top"
   >
     #### Fixed
-    The button bar is fixed at the bottom of the viewport when:
+    Fix the button bar at the bottom of the viewport when:
     - A user can proceed (or step back) regardless of form field completion.
-    - If form content overflows, the button bar is displayed with a secondary border style.
     - The form is in a wizard.
+    If form content overflows, the button bar uses a secondary border style.
```
Comment (origami-z) [inline site/docs/patterns/forms.mdx:417]: Similar here, 2nd bullet or not?

**Review by origami-z — DISMISSED** [review]
looks fine if bullets are intended

**Review by joshwooding — APPROVED** [review]

**Review by origami-z — APPROVED** [review]

**Pull request merged**